src/probnum/diffeq/_perturbsolve_ivp.py

```diff
@@ -14,45 +17,54 @@
 }
 """Implemented Scipy solvers."""
 
+
+# aliases for (otherwise too-)long lines
+lognorm = perturbed.step.PerturbedStepSolver.construct_with_lognormal_perturbation
+uniform = perturbed.step.PerturbedStepSolver.construct_with_uniform_perturbation
 PERTURBS = {
-    "step-lognormal": perturbed.step.PerturbedStepSolver.construct_with_lognormal_perturbation,
-    "step-uniform": perturbed.step.PerturbedStepSolver.construct_with_uniform_perturbation,
+    "step-lognormal": lognorm,
+    "step-uniform": uniform,
 }
 """Implemented perturbation-styles."""
 
 
+# This interface function is allowed to have many input arguments.
+# Having many input arguments implies having many local arguments,
+# so we need to disable both here.
+# pylint: disable="too-many-arguments,too-many-locals"
 def perturbsolve_ivp(
-    f,
-    t0,
-    tmax,
-    y0,
-    rng,
-    method="RK45",
-    perturb="step-lognormal",
-    noise_scale=10.0,
-    adaptive=True,
-    atol=1e-6,
-    rtol=1e-3,
-    step=None,
-    time_stops=None,
-):
-    r"""Solve an initial value problem with a perturbation-based probabilistic ODE solver.
+    f: Callable,
+    t0: FloatLike,
+    tmax: FloatLike,
+    y0: ArrayLike,
+    rng: np.random.Generator,
+    method: str = "RK45",
+    perturb: str = "step-lognormal",
+    noise_scale: FloatLike = 10.0,
+    adaptive: bool = True,
+    atol: FloatLike = 1e-6,
+    rtol: FloatLike = 1e-3,
+    step: Optional[FloatLike] = None,
+    time_stops: Optional[ArrayLike] = None,
+) -> perturbed.step.PerturbedStepSolution:
+    r"""Solve an initial value problem with a perturbation-based ODE solver.
 
     Parameters
     ----------
-    f :
+    f
         ODE vector field.
-    t0 :
+    t0
         Initial time point.
-    tmax :
+    tmax
         Final time point.
-    y0 :
+    y0
         Initial value.
-    rng :
+    rng
         Random number generator.
-    method :
+    method
         Integration method to use.
-        The following are available (docs adapted from `SciPy <https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html>`_):
+        The following are available
+        (docs adapted from `SciPy <https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html>`_):  # pylint: disable="line-too-long"
```

Comment:
this disable message here is a bit annoying, because it is rendered in the docs. I will try to fix it somehow.